packages/react-scripts/config/paths.js

```diff
@@ -85,3 +85,7 @@ if (__dirname.indexOf(path.join('packages', 'react-scripts', 'config')) !== -1)
     nodePaths: nodePaths
   };
 }
+
+configs.publicUrl = process.env.PUBLIC_URL ? process.env.PUBLIC_URL : require(configs.appPackageJson).homepage;
```

Comment:
I'd prefer we write `getPublicUrl()` and call it from every individual config block.